docs/proposals/api_design.md

```diff
@@ -0,0 +1,62 @@
+---
+title: "HTTP API Design"
+linkTitle: "HTTP API Design"
+weight: 1
+slug: http-api-design
+---
+
+# HTTP API Design
+
+## Overview
+
+The purpose of this design document is to propose a set of standards that should be the basis of the Cortex HTTP API. This document will outline the current state of the Cortex http api and describe limitations that result from the current approach. It will also outline a set of paradigms on how http routes should be created within Cortex.
+
+## Current Design
+
+As things currently stand, the majority of HTTP API calls exist under the `/api/prom` path prefix. This prefix is configurable. However, since this prefix is shared between all the modules which leads to conflicts if the Alertmanager is attempted to be run as as part of the single binary.
+
+## Proposed Design
```

Comment:
In the Cortex API endpoints there's also a mixed usage of `_` and `-`. I would suggest to uniform it (`-` looks more natural IMO).

Reply:
I'm not sure what you mean? I'm not aware on any endpoints with `-` in the path? Also since Prometheus uses `_` shouldn't we follow suit?

Reply:
Sure. My point was just about being consistent (and here being consistent with Prometheus `_`  looks perfect). I left the comment without double checking and I thought there was a mixed usage.

Reply:
👍